### ISSUE TITLE
Questions about JHU CSSE COVID-19 death data for NY

### ISSUE THREAD

**Issue by elray1** [cmu-delphi/covidcast-indicators#1321]
We've been spot checking the versioned data in covidcast and comparing to our systems, and we've turned up a few questions. I'm posting just one example here for now, in the hopes that it's a representative example of what's going on elsewhere: deaths in the state of NY with as of dates between August 27, 2020 and Sept 7, 2020.

First, here is a figure showing these data from our two systems, facetted by the as of date: 
![Screenshot from 2021-10-14 17-07-49](https://user-images.githubusercontent.com/5132208/137395654-ac901f67-0312-4483-8e47-1754a9229c73.png)

There are a few things worth noting here:
 1. The data from the two systems lined up well on Aug 27 (and for as of dates before that).
 2. For several days starting on Aug 28, it looks like covidcast may be catching a negative report for incidence and making some adjustments. Do you know what's going on? Was there an adjustment system like this in place at some point in the data ingestion or processing code in covidcast?
 3. A large outlier in reported deaths first shows up in our systems for different as of dates (2020-08-31 in our system, 2020-09-01 in covidcast) -- this is OK! We have different rules for when we pull commits and which commits are assigned to which as of dates. The commits were made at a borderline time. This 1 day difference is not an indication of a problem.
 4. The outlier is resolved one day after it shows up in our system, but it is not resolved until some time later in covidcast. I don't understand why this is.
 5. Note that around this time, JHU/CSSE was making some changes to their handling of the data in NY. Not sure how this would affect either of our systems.... See the commit history around that time [here](https://github.com/CSSEGISandData/COVID-19/commits/master?after=8ef2349c29cd33b418c5f3a806ef4b7f6cef6a75+769&branch=master&path%5B%5D=csse_covid_19_data&path%5B%5D=csse_covid_19_time_series&path%5B%5D=time_series_covid19_deaths_US.csv) and a relevant issue on their repo describing what they were doing [here](https://github.com/CSSEGISandData/COVID-19/issues/3084).

I'm pasting in some R code here to make the plot above for the covidcast data:
```
library(covidcast)
library(tidyverse)

covidcast_ny_data <- purrr::map_dfr(
  seq.Date(as.Date("2020-08-27"), by = 1, length.out = 12),
  function(as_of) {
    covidcast::covidcast_signal(
      data_source = "jhu-csse",
      signal = "deaths_incidence_num",
      geo_type = "state",
      geo_values = "ny",
      as_of = as_of
    ) %>%
      dplyr::mutate(as_of = as_of)
  }
)

ggplot() +
  geom_line(
    data = covidcast_ny_data,
    mapping = aes(x = time_value, y = value)
 ) +
 facet_wrap( ~ as_of)
```

I'm also pasting in some code (adapted from functions in the covidData package) to directly load data from the version of the data file committed to the CSSE repo on different dates, which more or less reproduces part of the plot showing data as covidData returns it:

```
# utility function to load data for NY from a csse time series file
load_ny_deaths_direct <- function(csse_path) {
  result <- readr::read_csv(csse_path) %>%
    tidyr::pivot_longer(
      matches("^\\d{1,2}\\/\\d{1,2}\\/\\d{2,4}$"),
      names_to = "date",
      values_to = "cum"
    ) %>%
    dplyr::mutate(
      date = as.character(lubridate::mdy(date))
    ) %>%
    dplyr::filter(Province_State == "New York") %>%
    dplyr::group_by(date) %>%
    dplyr::summarize(cum = sum(cum)) %>%
    dplyr::mutate(
      inc = diff(c(0, cum)),
      date = as.Date(date)
    ) %>%
    dplyr::ungroup()
  return(result)
}

# path to data file in CSSE repo, committed Aug 30, 2020, 12:50AM EDT
csse_path_as_of_2020_08_30 <- "https://raw.githubusercontent.com/CSSEGISandData/COVID-19/56f5b3ce5d18599ebcff7bbcaa0dd9842ae099ba/csse_covid_19_data/csse_covid_19_time_series/time_series_covid19_deaths_US.csv"

# path to data file in CSSE repo, committed Sep 7, 2020, 12:50AM EDT
csse_path_as_of_2020_09_07 <- "https://raw.githubusercontent.com/CSSEGISandData/COVID-19/6ed3d6d338344b8663d94e1e5e93ff39907ba923/csse_covid_19_data/csse_covid_19_time_series/time_series_covid19_deaths_US.csv"

csse_deaths_direct <- dplyr::bind_rows(
  load_ny_deaths_direct(csse_path_as_of_2020_08_30) %>%
    dplyr::mutate(as_of = "2020-08-30"),
  load_ny_deaths_direct(csse_path_as_of_2020_09_07) %>%
    dplyr::mutate(as_of = "2020-09-07")
) %>%
  dplyr::mutate(date = as.Date(date))

ggplot() +
  geom_line(
    data = csse_deaths_direct,
    mapping = aes(x = date, y = inc)
 ) +
 facet_wrap( ~ as_of)
```

**Comment by elray1** [comment]
OK, thanks! If you could let me know (by a slack message to be sure I see it) when you have implemented the fix for #1322, we'll re-run the checks that generated this issue and see if it resolves this and the other similar discrepancies we were seeing.

**Comment by krivard** [comment]
This really _should_ have been fixed by the patch we installed this week, but I'm not seeing the results in the database. We're checking a few things and may need to try again next week.
